packages/circuits/src/lib/BytePacking.circom

```diff
@@ -0,0 +1,58 @@
+pragma circom 2.0.0;
+
+include "../../node_modules/circomlib/circuits/comparators.circom";
+include "../../node_modules/circomlib/circuits/bitify.circom";
+
+/// @dev when dealing with Num2Bits in circom, LSB is stored in index 0, MSB in last index
+/// @dev therefore when converting from bits to numbers (Bits2Num) and vice versa (Num2Bits)
+/// @dev we need to first reverse the order of the cards
+template UnpackCards(n) {
+
+    signal input packedCards[n];
+    signal output unpackedCards[n*31];
+    
+    // unpack cards into bytes
+    component lt[n*31];
+    for (var i = 0; i < n; i++) {
+        var sum = 0;
+        var mult = 1;
+        for (var j = 0; j < 31; j++) {
+            unpackedCards[(i*31) + (30-j)] <-- (packedCards[i] >> (j*8)) & 255;
+            lt[(i*31)+j] = LessEqThan(8);
+            lt[(i*31)+j].in[0] <== unpackedCards[(i*31) + (30-j)];
```

Comment:
Both the indexes `i*31+j` and `(i*31) + (30-j)` are reused throughout, so could be good to make them variables with meaningful names.

Actually, why not reuse `(i*31) + (30-j)` for the `lt` index?

Reply:
Would also reintroduce the comment that says that because we are comparing `< 255`, 255 is not a valid value (which will be relevant if we shift the null value to be 0 later)

Reply:
Good points!